File: content/problems/2442-count-number-of-distinct-integers-after-reverse-operations/solutions/set-original-and-reversed-values/Solution.java
class Solution {
    public int countDistinctIntegers(int[] nums){
        Set<Integer> values=new HashSet<>();
        for(int value:nums){
            values.add(value);
            int x=value,reversed=0;
            while(x>0){
                reversed=reversed*10+x%10;
                x/=10;
            }
            values.add(reversed);
        }
        return values.size();
    }
}

File: content/problems/2442-count-number-of-distinct-integers-after-reverse-operations/solutions/set-original-and-reversed-values/Solution.kt
class Solution {
    fun countDistinctIntegers(nums:IntArray):Int{
        val values=HashSet<Int>();
        for(value in nums){
            values.add(value);
            var x=value;
            var reversed=0;
            while(x>0){
                reversed=reversed*10+x%10;
                x/=10
            };
            values.add(reversed)
        };
        return values.size
    }
}
